src/code/code_800EC960.c

```diff
@@ -1203,916 +1233,1080 @@ u8 D_8016B9F4;
-                sCurOcarinaSong[sOcarinaSongAppendPos - 1] = sCurOcarinaBtnVal;
+        if (sOcarinaWithoutMusicStaffPos == 8) {
+            for (i = 0; i < 7; i++) {
+                sCurOcarinaSongWithoutMusicStaff[i] = sCurOcarinaSongWithoutMusicStaff[i + 1];
             }
+        } else {
+            sOcarinaWithoutMusicStaffPos++;
+        }
 
-            for (i = sOcarinaSongNoteStartIdx; i < sOcarinaSongCnt; i++) {
-                if (sOcarinaAvailSongs & (u16)(1 << i)) {
-                    for (j = 0, k = 0;
-                         j < gOcarinaSongNotes[i].len && k == 0 && sOcarinaSongAppendPos >= gOcarinaSongNotes[i].len;) {
-                        temp_v0 = sCurOcarinaSong[(sOcarinaSongAppendPos - gOcarinaSongNotes[i].len) + j];
-                        if (temp_v0 == sOcarinaNoteValues[gOcarinaSongNotes[i].notesIdx[j]]) {
-                            j++;
-                        } else {
-                            k++;
-                        }
-                    }
+        if (ABS_ALT(sCurOcarinaBendIndex) > 20) {
+            sCurOcarinaSongWithoutMusicStaff[sOcarinaWithoutMusicStaffPos - 1] = OCARINA_PITCH_NONE;
+        } else {
+            sCurOcarinaSongWithoutMusicStaff[sOcarinaWithoutMusicStaffPos - 1] = sCurOcarinaPitch;
+        }
 
-                    if (j == gOcarinaSongNotes[i].len) {
-                        D_80131878 = i + 1;
-                        sOcarinaInpEnabled = 0;
-                        D_80130F3C = 0;
+        // This nested for-loop tests to see if the notes from the ocarina are identical
+        // to any of the songIndex from sFirstOcarinaSongIndex to sLastOcarinaSongIndex
+
+        // Loop through each of the songs
+        for (i = sFirstOcarinaSongIndex; i < sLastOcarinaSongIndex; i++) {
+            // Checks to see if the song is available to be played
+            if (sAvailOcarinaSongFlags & (u16)(1 << i)) {
+                for (j = 0, k = 0; j < gOcarinaSongButtons[i].numButtons && k == 0 &&
+                                   sOcarinaWithoutMusicStaffPos >= gOcarinaSongButtons[i].numButtons;) {
+                    pitch = sCurOcarinaSongWithoutMusicStaff[sOcarinaWithoutMusicStaffPos -
+                                                             gOcarinaSongButtons[i].numButtons + j];
+                    if (pitch == sButtonToPitchMap[gOcarinaSongButtons[i].buttonsIndex[j]]) {
+                        j++;
+                    } else {
+                        k++;
                     }
                 }
+
+                // This conditional is true if songIndex = i is detected
+                if (j == gOcarinaSongButtons[i].numButtons) {
+                    sPlayedOcarinaSongIndexPlusOne = i + 1;
+                    sIsOcarinaInputEnabled = false;
+                    sOcarinaFlags = 0;
+                }
             }
         }
     }
 }
 
-void func_800ED458(s32 arg0) {
-    u32 phi_v1_2;
+void AudioOcarina_PlayControllerInput(u8 unused) {
```

Comment:
This is an interesting case. We very likely know what this argument is and its type due to MM, because the function is near identical in MM but this arg is used as a u8 bool:

https://github.com/engineer124/mm/blob/a501e8ca1363530e1127b2093451950f7b8f7386/src/code/audio/audio_interface.c#L2375

Reply:
Alright, maybe this should have a comment then to explain the u8 like `This function takes a u8 in Majora's Mask`